### PR TITLE
Add rif tests

### PR DESCRIPTION
**Pull request by BonfaceKilz**
### Description

This PR introduces RDF unit test cases for the RIF fetches.  For more context read [gn3/ADR-000](https://issues.genenetwork.org/topics/ADR/gn3/000-add-test-cases-for-rdf).

### Testing

- Make sure you have a local virtuoso server running.
- Make sure you set the following in your secrets file:
 ```
SPARQL_USER = "XXX"
SPARQL_PASSWORD = "XXXXX"
SPARQL_AUTH_URI="http://localhost:9082/sparql-auth/"
SPARQL_CRUD_AUTH_URI="http://localhost:9082/sparql-graph-crud-auth"
 ```
 - Run tests by: `pylint -k rdf`
 
 TODO:
- Update the CD linux container with above params.

### CODE REVIEW

File: tests/fixtures/rdf.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from flask import config
+import os
+import requests
+import pytest
+
+
+def get_sparql_auth_conf() -> dict:
+    """Fetch SPARQL auth configuration for the GN3_SECRETS file."""
+    sparql_conf = config.Config("")
+    sparql_conf.from_envvar("GN3_SECRETS")
```
Comment (jnduli) [inline tests/fixtures/rdf.py]: I think these should be reorderd i.e. GN3_CONF takes precedence over GN3_SECRETS

File: gn3/db/rdf/wiki.py [hunk 2]
```diff
@@ -43,6 +43,7 @@ def __sanitize_result(result: dict) -> dict:
     categories = result.get("categories")
```
Comment (jnduli) [inline gn3/db/rdf/wiki.py:43]: minor: we could improve this by:
```
categories = result.get("categories", [])
if isinstance(categories, str):
  result["categories"] = [categories]
```

File: README.md [hunk 3]
```diff
@@ -126,11 +126,24 @@ These configurations should be set in an external config file, pointed to with t
...
+
+```
+SPARQL_USER = "XXX"
+SPARQL_PASSWORD = "XXXXXXXXXX"
+SPARQL_AUTH_URI="http://localhost:9082/sparql-auth/"
+SPARQL_CRUD_AUTH_URI="http://localhost:9082/sparql-graph-crud-auth"
+FAHAMU_AUTH_TOKEN="XXXXXX"
+```
+
+Note: The sparql configurations are important for running tests I.e. `pytest -k rdf`.
```
Comment (jnduli) [inline README.md:145]: Add a link to https://issues.genenetwork.org/topics/systems/virtuoso to help set up virtuoso

File: README.md [hunk 4]
```diff
@@ -126,11 +126,24 @@ These configurations should be set in an external config file, pointed to with t
...
 TMPDIR also needs to be set correctly for the R script(s) because they pass results on as files on the local system (previously there was an issue with it being set to /tmp instead of ~/genenetwork3/tmp). Note that the Guix build system should take care of the paths.
 
+### Secrets
+
+All of GN3's secret parameters are found inside the "GN3_SECRETS".  This file should contain the following:
+
+```
+SPARQL_USER = "XXX"
+SPARQL_PASSWORD = "XXXXXXXXXX"
+SPARQL_AUTH_URI="http://localhost:9082/sparql-auth/"
```
Comment (jnduli) [inline README.md]: can we change this to the default port virtuoso provides?

**Review by jnduli — APPROVED** [review]

**Comment by Alexanderlacuna** [comment]
LGTM!

**Pull request merged**